src/timetable.h

```diff
@@ -16,6 +16,40 @@
 
 static const TimerGameCalendar::Year MAX_TIMETABLE_START_YEARS = 15; ///< The maximum start date offset, in years.
 
+/**
+ * Get the TimerGameTick::TickCounter tick of a given date.
+ * @param start_date The date when the timetable starts.
+ * @return The first tick of this date.
+ */
+inline TimerGameTick::TickCounter GetStartTickFromDate(TimerGameCalendar::Date start_date)
+{
+	/* Calculate the offset in ticks from the current date. */
+	TimerGameTick::Ticks tick_offset = (start_date - TimerGameCalendar::date).base() * Ticks::DAY_TICKS;
+
+	/* Compensate for the current date_fract. */
+	tick_offset -= TimerGameCalendar::date_fract;
+
+	/* Return the current tick plus the offset. */
+	return TimerGameTick::counter + tick_offset;
+}
+
+/**
+ * Get a date from a given start tick of timetable.
+ * @param start_tick The TimerGameTick::TickCounter when the timetable starts.
+ * @return The date when we reach this tick.
+ */
+inline TimerGameCalendar::Date GetDateFromStartTick(TimerGameTick::TickCounter start_tick)
+{
+	/* Calculate the offset in ticks from the current counter tick. */
+	TimerGameTick::Ticks tick_offset = start_tick - TimerGameTick::counter;
+
+	/* Compensate for the current date_fract. */
+	tick_offset += TimerGameCalendar::date_fract;
+
+	/* Return the current date plus the offset in days. */
+	return TimerGameCalendar::date + (tick_offset / Ticks::DAY_TICKS);
+}
```

Comment:
Do these need to be inline functions? I think splitting definition and declaration is fine for these.